### ISSUE TITLE
iOS - Appium is not able to properly install then open app in iOS simulator

### ISSUE THREAD

**Issue by kingages** [appium/appium#16050]
## The problem

We are using appium with iOS simulator and it is kind of random issue. We are facing issue that installation is not properly finished. Sometimes it is ok but in other cases it ends with error like below.

- We had no issue like that before appium update from 1.21 to 1.22. Xcode 12.4 - iOS 14.4 -> Xcode 13 - iOS 15.0

![image](https://user-images.githubusercontent.com/26042586/140712496-7dd774b7-2867-4b0c-ad45-6939bbc21ddb.png)

Does anyone experience issue like this?

## Environment
* Appium 1.22 previously we had appium 1.21
* Selenium Grid hub (selenium-server-standalone-3.141.59) with appium node and all is running on one device
* macOS Big Sur - Xcode 13 and iOS 15.0 - simulator

## Details

Let's say we one test case that is executed every two hours and three times it is failed then suddenly it is passed. Test case is very simple install and open app in iOS simulator.

Also if I restart the appium node that is connected to grid hub issue is fixed for while.

## Appium logs
[Logs are just consequence of the issue it is not root cause message](https://gist.github.com/kingages/a800e205513339d862a4ee6f8a507be8).

**Comment by mykola-mokhnach** [comment]
It looks like an issue with the simulator itself. Appium simple calls `xcrun simctl` to perform app install.

**Comment by kingages** [comment]
I tried to use absolute path to the same build and it works with absolute path but not with URL. Issue seems to be related to way how zip file is downloaded from URL.

**Comment by kingages** [comment]
```
dbug Support https://artifacts.merck.com/artifactory/list/generic-smart-dev-local/BUILDS/ROBOT/MVR-DEV-APP-latest.zip (30.49 MB) has been downloaded to '/var/folders/4z/vzvd16n904vghmqfp5yc624m0000gr/T/2021108-27475-1uohkmy.zt0k/MVR-DEV-APP-latest.zip' in 3.184s
dbug Support Approximate download speed: 9.57 MB/s
```
For some reason if I delete folders with artefacts in /var/folders/4z/vzvd16n904vghmqfp5yc624m0000gr/T/. It works again.

**Comment by mykola-mokhnach** [comment]
The attached log has not enough details to make any assumptions, but in case multiple tests are executed at the same time there could be some race conditions. Usually this type of issues is quite hard to track down and debug, so I am not quite sure we would be able to help much there without knowing where exactly the race happens.

https://github.com/appium/appium-base-driver/blob/44368298390ca4387121b132358d8725f796fa41/lib/basedriver/helpers.js#L124 could be a starting point for you to debug it.

**Comment by kingages** [comment]
Looks like it is connected with change https://github.com/appium/appium-support/pull/226 that is touching unzip function.

**Comment by mykola-mokhnach** [comment]
Everything could happen. Try to find out what is the difference between unzipping your archive with the system unzip vs the built-in one

**Comment by nagyzsolt1989** [comment]
We are experiencing quite similar intermittent install issues since we upgraded to 1.22.0

I checked and `DummyAPP MFA.app/Frameworks/Protobuf.framework/Info.plist` file is definitely there, but it seems like Appium couldn't reach it.

[Authenticator_install.log](https://github.com/appium/appium/files/7506133/Authenticator_install.log)

**Comment by jlipps** [comment]
@nagyzsolt1989 what happens if you name your app without a space in it.

**Comment by KazuCocoa** [comment]
You could https://github.com/appium/appium-base-driver/blob/bb5a321d2c0b7fc6146e8f8ffec0804394ea4354/lib/basedriver/helpers.js#L331-L333 as false to check if unzip change could be relevant.

**Comment by nagyzsolt1989** [comment]
Thanks for the quick replies!

@jlipps Unfortunately the problem also occurs without a space in the app name

@KazuCocoa I will set the useSystemUnzip to false and post the result soon

**Comment by nagyzsolt1989** [comment]
I haven't experienced the issue since we set useSystemUnzip to false (but it might be also because of the intermittent nature of this issue).

I'll monitor our runs for a couple of days and get back to you if it will occur.

**Comment by kingages** [comment]
I'm also trying @KazuCocoa suggestion but as was said it needs to be monitored.

**Comment by KazuCocoa** [comment]
Thank you so much

**Comment by mykola-mokhnach** [comment]
I've added `APPIUM_PREFER_SYSTEM_UNZIP` environment variable support to appium@beta. By setting this variable to `0` or `false` drivers will always use the internal unzip module instead of the system one.

**Comment by nagyzsolt1989** [comment]
It looks like that we haven't had the issue since the change. Thanks for the suggestion!

**Comment by KazuCocoa** [comment]
Okay, thank you for the confirmation.
This will be included in 1.22.1 https://github.com/appium/appium/pull/16002

**Comment by kingages** [comment]
It works for us too. Thank you very much.
